nexus/social_media/services.py

```diff
@@ -4,16 +4,167 @@
 
 # Third Party Stuff
 import facebook
+import requests
 import tweepy
 from django.conf import settings
 from django.utils import timezone
+from rest_framework import status
 
 # nexus Stuff
-from nexus.base import exceptions
+from nexus.base import exceptions as exc
 from nexus.social_media import tasks
 from nexus.social_media.models import Post
 
 
+def upload_image_to_linkedin(author, headers, post, linkedin_auth):
+    """Upload image to linkedin for given post.
+
+    :param author: Owner of the post to be used for response POST data.
+
+    :param headers: HTTP headers required for successful HTTP request.
+
+    :param post: Post model instance.
+
+    :param linkedin_auth: LINKEDIN_AUTH object taken from settings.
+
+    :returns: value of asset
+
+    """
+    post_data_assets = {
+        'registerUploadRequest': {
+            'recipes': [
+                'urn:li:digitalmediaRecipe:feedshare-image'
+            ],
+            'owner': author,
+            'serviceRelationships': [
+                {
+                    'relationshipType': 'OWNER',
+                    'identifier': 'urn:li:userGeneratedContent'
+                }
+            ]
+        }
+    }
+
+    api_url_assets = f'{settings.LINKEDIN_API_URL_BASE}assets?action=registerUpload'
+
+    response_assets = requests.post(api_url_assets, json=post_data_assets,
+                                    headers=headers)
+    if response_assets.status_code == status.HTTP_200_OK:
+        json_response_assets = response_assets.json()
+
+        value = json_response_assets.get('value')
+        upload_mechanism = value.get('uploadMechanism')
+        upload_http_request = upload_mechanism.get('com.linkedin.digitalmedia.uploading.MediaUploadHttpRequest')
+        upload_url = upload_http_request.get('uploadUrl')
+
+        asset = value.get('asset')
+
+        response_upload_image = requests.post(upload_url,
+                                              data=post.image.file.read(),
+                                              headers={'Authorization': f"Bearer {linkedin_auth['access_token']}"})
+        if response_upload_image.status_code == status.HTTP_201_CREATED:
+            return asset
+        else:
+            appropriate_response_action(response_upload_image)
```

Comment:
I would not use this function and get rid of this. I would actually raise an error from here.

Also, for that function, what happens if you somehow give `200` status code to it? :)